app/models/piece.rb

```diff
@@ -1,111 +1,27 @@
 class Piece < ActiveRecord::Base
   belongs_to :game
 
-	def is_obstructed?(x, y)
-
-		#check moving direction
-		path = moving_direction(x, y)
-
-		#path is diagonal from lower left to upper right
-		if path = 'diagonal' && x_cord < x && y_cord < y
-			(x_cord + 1).upto(x - 1) do |changing_x|
```

Comment:
@ilyakrasnov this is the original obstructed method written by @catherinezzy. 

is upto and downto doing what we want it to do??? Is it really iterating each and passing it through out occupied? method? is our occupied method written correctly? 

Even if the "path" variable or the moving_direction method is not correct, which I've rewritten this code without like 20 different times, I can't figure out how to iterate through each of the x & y coords to a method that checks whether there is a piece in that coord.  

It sounds so logical, but I keep making a wrong turn into a dead end somewhere. I know it doesn't work because the tests don't pass for horizontal moves. But maybe we've approached writing the test wrong? I'll go comment there now.